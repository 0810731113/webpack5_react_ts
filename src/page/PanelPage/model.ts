export interface QTResponse<T = {}> {
  actionname: string;
  response: {
    status: "ok" | "error";
    data: T;
  };
  arguments?: T;
}

export interface QTErrorMessage {
  errormessage: string;
}

export interface QTRequest {
  actionname: string;
  arguments?: {};
}

export interface LoginResponsePayload {
  accessToken: string;
  userId: string;
  userName: string;
}

export interface QTErrorData {
  errormessage: string;
  errornumber?: number;
}
