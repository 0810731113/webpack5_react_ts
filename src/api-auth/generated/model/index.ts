export * from './account-vo';
export * from './department';
export * from './enterprise-account';
export * from './enterprise-account-creation';
export * from './enterprise-account-creation-v2';
export * from './enterprise-account-query';
export * from './enterprise-info';
export * from './enterprise-member';
export * from './mobile-bind-account';
export * from './oauth2-access-token';
export * from './pagination-enterprise-member';
export * from './personal-account-creation';
export * from './response-account-vo';
export * from './response-enterprise-account';
export * from './response-enterprise-info';
export * from './response-list-department';
export * from './response-list-mobile-bind-account';
export * from './response-pagination-enterprise-member';
export * from './response-user-info';
export * from './response-user-info-with-token';
export * from './response-void';
export * from './responseboolean';
export * from './user-info';
export * from './user-info-with-token';
export * from './user-login';
