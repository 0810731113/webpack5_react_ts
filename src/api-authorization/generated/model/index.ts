export * from './resource-permission';
export * from './response-list-resource-permission';
export * from './response-list-user-roles';
export * from './response-liststring';
export * from './response-void';
export * from './user-role-creation';
export * from './user-roles';
