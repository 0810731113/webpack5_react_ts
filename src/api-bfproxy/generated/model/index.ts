export * from './bimface-upload-body';
export * from './bimface-upload-result';
export * from './condition';
export * from './data-set-integrate-bean';
export * from './data-set-integrate-request';
export * from './link';
export * from './model-and-view';
export * from './responsestring';
export * from './view';
