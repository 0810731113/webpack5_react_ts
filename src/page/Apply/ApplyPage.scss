@import "../../style/colors.scss";

.apply-enter {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex: auto;

  & > .ant-row {
    flex: auto;

    & > .ant-col {
      display: flex;

      & > a {
        flex: auto;
      }
    }
  }

  .apply-enter-card {
    text-align: center;
    height: 50vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    flex: auto;
    min-height: 400px;
    width: 100%;

    box-shadow: 0 1px 6px 1px rgba(0, 0, 0, 0.1);

    .antd-card-body {
      padding: 42px 52px;
    }

    .ant-typography {
      font-size: 12px;
      line-height: 2.33;
      color: $text-color-secondary;
      margin: 0;
      margin-bottom: 36px;
    }

    h3.ant-typography {
      font-size: 20px;
      line-height: 1.4;
      color: $text-color-important;
      margin-bottom: 24px;
    }

    .enter-button {
      width: 200px;
      height: 54px;

      font-size: 20px;
    }

    .enter-icon {
      width: 110px;
      height: 110px;
      margin-bottom: 15px;
    }
  }

  a + a {
    margin-left: 32px;
  }
}

.apply-layout-page {
  min-height: 100%;
  background-color: #f5f5f5;

  display: flex;
  flex-direction: column;

  .page-header {
    flex: 0;
  }

  .apply-page {
    flex: 1;
  }

  .footer {
    flex: 0;
    text-align: center;
    margin: 20px 0;

    .copyright {
      color: rgba(0, 0, 0, 0.65);
      font-size: 12px;
      line-height: 1.67;

      a {
        color: rgba(0, 0, 0, 0.65);
        margin-right: 3px;

        &:hover {
          color: rgba(0, 0, 0, 0.85);
        }
      }
    }
  }
}

.apply-page {
  display: flex;
  flex-direction: column;

  align-items: center;

  .header {
    height: 140px;
    //border-bottom: 1px solid lightgrey;
    width: 100%;

    position: relative;

    // background-image: url("/web/main_banner.png");
    background-repeat: round;
    //background-size: 100%, 140px;

    display: flex;
    align-items: center;
    justify-content: center;

    .slogon {
      text-align: center;
      color: white;
      z-index: 10;

      .main {
        font-size: 16px;
        line-height: 1.5;
        margin-bottom: 8px;
      }

      .sub {
        opacity: 0.6;
        line-height: 2;
      }
    }

    .steps {
      margin: auto;
      width: 1000px;

      .ant-steps-item-container > .ant-steps-item-content {
        .ant-steps-item-title {
          color: white;

          &::after {
            background-color: white;
          }
        }

        .ant-steps-item-description {
          color: white;
        }
      }

      .ant-steps-item-finish,
      .ant-steps-item-wait {
        .ant-steps-item-icon {
          background-color: transparent;
          border-color: white;
          color: white;

          .ant-steps-icon {
            color: white;
          }
        }
      }

      .ant-steps-item-description {
        max-width: none;
      }
    }
  }

  .exclamation {
    margin-bottom: 16px;
    color: rgba(0, 0, 0, 0.65);
    text-align: justify;
    font-size: 12px;
  }

  .content {
    margin-top: 64px;
    width: 400px;

    form + form {
      margin-top: 16px;
    }

    button {
      width: 100%;
    }

    .ant-result-icon {
      margin-bottom: 10px;

      .anticon {
        color: #13c2c2;
      }
    }

    .ant-result-title {
      color: #13c2c2;
      font-size: 36px;
    }

    .ant-result-extra {
      margin-top: 10px;
      color: rgba(0, 0, 0, 0.45);
      font-size: 12px;
    }

    .commit-wrap {
      display: flex;
      flex-direction: column;
      margin-top: 16px;
    }
  }
}

@media screen and (max-width: 500px) {
  .apply-enter {
    & > .ant-row {
      & > .ant-col {
        flex-direction: column;
        padding: 0 32px;

        .apply-enter-card {
          height: 240px;

          .enter-icon {
            width: 64px;
            height: 64px;
          }

          min-height: auto;

          .enter-button {
            display: none;
          }

          .ant-typography {
            margin-bottom: 8px;
          }
        }

        a + a {
          margin-left: 0;
          margin-top: 32px;
        }
      }
    }
  }

  .apply-layout-page {
    overflow: scroll;

    &.succeeded {
      .apply-page {
        min-height: calc(100vh - 84px);
      }
    }

    .page-header {
      padding: 7.73vw 0 0 6.4vw;
      background: transparent;
      box-shadow: none;
      height: auto !important;
      min-height: auto !important;
      max-height: none !important;
      flex: none;

      &.succeeded {
        position: absolute;
        bottom: 20px;
        left: 50%;
        transform: translateX(-50%);
        padding: 0;

        .ant-space.ant-space-horizontal.ant-space-align-center {
          img {
            margin-left: 0;
          }
        }
      }

      .ant-space.ant-space-horizontal.ant-space-align-center {
        flex-direction: column;

        img {
          height: 30px !important;
          margin-bottom: 56.5px;
          // margin-left: -40px;
        }
      }
    }

    .apply-page {
      flex: none;
      min-height: calc(100vh - 250px);

      .header {
        display: none;
      }

      .content {
        margin-top: 36px;
        width: 87.2vw;

        &.succeeded {
          margin-top: calc(50vh - 60px - 144px);
        }

        .ant-input,
        .ant-input-affix-wrapper,
        .ant-select-selector,
        .ant-btn {
          height: 47px;
          border-radius: 5px !important;

          .ant-select-selection-item {
            line-height: 45px;
          }
        }

        .ant-input-affix-wrapper {
          .ant-input {
            height: 33px;
          }
        }

        .ant-form-item-label.ant-form-item-label-left {
          display: none;
        }

        .commit-wrap {
          flex-direction: column-reverse;

          .ant-row.ant-form-item {
            margin-top: 16px;
          }
        }

        .exclamation {
          font-size: 12px;
        }
      }
    }
  }
}
