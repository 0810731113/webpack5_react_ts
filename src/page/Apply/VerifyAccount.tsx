import React, { useEffect, useContext } from "react";
import { Switch, Route, useRouteMatch } from "react-router";
import { Link } from "react-router-dom";
import { useImmer } from "use-immer";
import { Form, Input, Button, Radio, Select, message } from "antd";
import { applyService } from "service";
import { onResponseError } from "function/auth.func";
import { ApplyPageContext } from "./ApplyPage";

interface VerifyAccountProps {
  onNext: () => void;
}

interface State {
  useNew: boolean;
}

export default function VerifyAccount(props: VerifyAccountProps) {
  const { onNext } = props;
  const [{ useNew }, updateState] = useImmer<State>({
    useNew: true,
  });

  const {
    accounts,
    applyingAccount,
    setIsNew,
    setApplyingAccountInfo,
  } = useContext(ApplyPageContext);

  useEffect(() => {
    updateState((draft) => {
      draft.useNew = !(accounts.length > 0);
    });
  }, [accounts]);

  const goNext = (isNew: boolean) => {
    setIsNew(isNew);
    onNext();
  };

  return (
    <>
      {accounts.length > 0 && (
        <div>
          <div style={{ marginBottom: 8 }}>
            检测到您已有广联云企业账号，是否使用现有账号？
          </div>
          <div style={{ marginBottom: 24 }}>
            <Radio.Group
              onChange={(e) => {
                updateState((draft) => {
                  draft.useNew = e.target.value;
                });
              }}
              value={useNew}
            >
              <Radio value={false}>使用现有账号</Radio>
              <Radio value>创建新账号</Radio>
            </Radio.Group>
          </div>
        </div>
      )}
      {useNew ? (
        <Form
          labelCol={{ span: 6 }}
          wrapperCol={{ span: 18 }}
          name="basic"
          labelAlign="left"
          initialValues={{ remember: true }}
          validateTrigger="onBlur"
          onFinish={(values) => {
            applyService
              .registerUserCenter(
                values.identity,
                values.password,
                applyingAccount?.passwordMobile ?? "",
              )
              .then(() => {
                setApplyingAccountInfo("identity", values.identity);
                setApplyingAccountInfo("password", values.password);
                goNext(true);
              })
              .catch(onResponseError);
          }}
          onFinishFailed={(errorInfo) => {
            console.log("Failed:", errorInfo);
          }}
        >
          <Form.Item
            label="账号名"
            name="identity"
            rules={[
              { required: true, message: "请输入账号" },
              {
                pattern: /^[A-Za-z\u4e00-\u9fa5][\w\u4e00-\u9fa5]{2,19}$/,
                message:
                  "3-20字符，需要以汉字或字母开头，由汉字、字母、数字或下划线组成",
              },
            ]}
          >
            <Input placeholder="请设置账号名" />
          </Form.Item>

          <Form.Item
            name="password"
            label="密码设置"
            rules={[
              {
                required: true,
                message: "请输入密码",
              },
              {
                pattern: /^(?!\d+$)(?![A-Za-z]+$)(?!([^\d()A-Za-z]|[()\\])+$)([^\d()A-Za-z]|[()\\]|[A-Za-z]|\d){8,16}$/,
                message: "请输入8-16位密码，数字、字母和符号至少包含两种",
              },
            ]}
            hasFeedback
          >
            <Input.Password placeholder="请设置密码" />
          </Form.Item>

          <Form.Item
            name="confirm"
            label="二次确认"
            dependencies={["password"]}
            hasFeedback
            rules={[
              {
                required: true,
                message: "请确认你的密码",
              },
              ({ getFieldValue }) => ({
                validator(rule, value) {
                  if (!value || getFieldValue("password") === value) {
                    return Promise.resolve();
                  }
                  // eslint-disable-next-line prefer-promise-reject-errors
                  return Promise.reject("两次密码输入不一致");
                },
              }),
            ]}
          >
            <Input.Password placeholder="请再次输入密码" />
          </Form.Item>

          <Form.Item wrapperCol={{ span: 24 }}>
            <Button type="primary" htmlType="submit">
              下一步
            </Button>
          </Form.Item>
        </Form>
      ) : (
        <Form
          labelCol={{ span: 6 }}
          wrapperCol={{ span: 18 }}
          name="basic"
          labelAlign="left"
          initialValues={{ remember: true }}
          validateTrigger="onBlur"
          onFinish={(values) => {
            applyService
              .verifyUserPassword(values.username, values.password)
              .then((account) => {
                setApplyingAccountInfo("identity", values.username);
                setApplyingAccountInfo("password", values.password);
                setApplyingAccountInfo("passwordEmail", account?.email ?? "");
                setApplyingAccountInfo("userName", account?.nickname ?? "");
                setApplyingAccountInfo("qq", account?.qq ?? "");
                setApplyingAccountInfo("company", account?.company ?? "");
                goNext(false);
              })
              .catch(() => {
                message.error("密码错误");
              });
          }}
          onFinishFailed={(errorInfo) => {
            console.log("Failed:", errorInfo);
          }}
        >
          <Form.Item
            label="现有账号"
            name="username"
            initialValue={accounts[0]?.username}
            rules={[
              {
                required: true,
                message: "请确认你的账号",
              },
            ]}
          >
            <Select>
              {accounts?.map((account) => (
                <Select.Option
                  value={account.username ?? "error"}
                  key={account.id}
                >
                  {account.username}
                </Select.Option>
              ))}
            </Select>
          </Form.Item>

          <Form.Item
            name="password"
            label="密码验证"
            rules={[
              {
                required: true,
                message: "请输入密码",
              },
            ]}
          >
            <Input.Password placeholder="请输入密码" />
          </Form.Item>

          <Form.Item wrapperCol={{ span: 24 }}>
            <Button type="primary" htmlType="submit">
              下一步
            </Button>
          </Form.Item>
        </Form>
      )}
    </>
  );
}
