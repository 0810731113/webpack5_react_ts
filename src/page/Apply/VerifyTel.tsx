import React, {
  useEffect,
  useContext,
  forwardRef,
  ForwardRefRenderFunction,
} from "react";
import { Switch, Route } from "react-router";
import { Link, useHistory } from "react-router-dom";
import { ExclamationCircleFilled } from "@ant-design/icons";
import { useImmer } from "use-immer";
import {
  Form,
  Input,
  Button,
  Checkbox,
  message,
  Modal,
  FormInstance,
} from "antd";
import { applyService } from "service";
import { onResponseError } from "function/auth.func";
import { MobileBindAccount } from "api-auth/generated/model";
import consts from "consts";
import { IsPC } from "function/common.func";
import { AxiosError } from "axios";
import { ApplyPageContext } from "./ApplyPage";

const { ENV } = consts;

interface VerifyTelProps {
  onNext?: () => void;
}

interface State {
  second: number;
}
const VerifyTel: ForwardRefRenderFunction<FormInstance<any>, VerifyTelProps> = (
  props: VerifyTelProps,
  ref: any | null,
) => {
  const { onNext } = props;
  const [{ second }, updateState] = useImmer<State>({ second: 0 });
  const { replace } = useHistory();

  const [form] = Form.useForm();
  if (ref) {
    ref.current = form;
  }
  useEffect(() => {
    if (second > 0) {
      setTimeout(() => {
        updateState((draft) => {
          draft.second -= 1;
        });
      }, 1000);
    }
  }, [second]);
  const { setAccounts, setApplyingAccountInfo } = useContext(ApplyPageContext);

  const sendSMS = async () => {
    const values = await form.validateFields(["mobile"]);
    try {
      if (onNext) {
        await applyService.sendSMS(values.mobile);
      } else {
        await applyService.getCode(values.mobile);
      }
      message.success("验证码已发送");
      updateState((draft) => {
        draft.second = 60;
      });
    } catch (error) {
      if (!onNext && error.response?.data.code === 50015) {
        Modal.confirm({
          icon: <ExclamationCircleFilled />,
          centered: true,
          title: "温馨提醒",
          content: `该手机号已申请过广联达数字设计试用资格，是否立即前往协同设计平台?`,
          okText: `立即前往`,
          onOk() {
            window.location.href = "/";
          },
        });
      } else {
        onResponseError(error);
      }
    }
  };

  const goNext = (
    accounts: MobileBindAccount[] | undefined,
    mobile: string,
  ) => {
    if (accounts instanceof Array && accounts.length > 0) {
      setAccounts(accounts.filter((acc) => acc?.status === -1));
    }
    setApplyingAccountInfo("passwordMobile", mobile);
    onNext?.();
  };

  const onSubmit = (mobile: string, code: string) => {
    applyService
      .getMainAccountList(mobile, code)
      .then((mobileAccounts) => {
        const appliedNames = mobileAccounts
          ?.filter((acc) => acc.status! >= 0)
          .map((acc) => acc.username ?? "");

        const unAppliedAccounts = mobileAccounts?.filter(
          (acc) => acc.status! >= -1,
        );

        if (appliedNames && appliedNames.length > 0) {
          // 测试专用代码
          if (ENV === "qa") {
            Modal.confirm({
              title: `温馨提示`,
              content: `该手机号已申请过试用资格，账号名为 ${appliedNames.join(
                ",",
              )}`,
              okText: "前往登录",
              cancelText: "继续（QA测试专用）",
              onOk() {
                replace("/enterprise");
              },
              onCancel() {
                goNext(unAppliedAccounts, mobile);
              },
            });
          } else {
            Modal.info({
              title: `温馨提示`,
              content: `该手机号已申请过试用资格，账号名为 ${appliedNames.join(
                ",",
              )}`,
              okText: IsPC() ? "前往登录" : "确定",
              onOk() {
                if (IsPC()) {
                  replace("/enterprise");
                }
              },
            });
          }
        } else {
          goNext(unAppliedAccounts, mobile);
        }
      })
      .catch(onResponseError);
  };

  return (
    <>
      <Form
        labelCol={{ span: 5 }}
        wrapperCol={{ span: 19 }}
        labelAlign="left"
        name="basic"
        form={form}
        validateTrigger="onBlur"
        onFinish={(values) => {
          // console.log("Success:", values);
          onSubmit(values.mobile, values.code);
        }}
        onFinishFailed={(errorInfo) => {
          console.log("Failed:", errorInfo);
        }}
      >
        <Form.Item
          label="手机号码"
          name="mobile"
          rules={[
            {
              pattern: /^1[3-9]\d{9}$/,
              message: "无效的手机号码",
            },
            { required: true, message: "请输入手机号码" },
          ]}
        >
          <Input size="large" placeholder="请输入手机号码" />
        </Form.Item>

        <Form.Item
          label="验证码"
          name="code"
          rules={[{ required: true, message: "请输入验证码" }]}
        >
          <Input
            size="large"
            suffix={
              <Button type="link" disabled={second > 0} onClick={sendSMS}>
                {second > 0 ? `${second}s` : "获取验证码"}
              </Button>
            }
            placeholder="请输入验证码"
          />
        </Form.Item>

        <div className="commit-wrap">
          <Form.Item
            // wrapperCol={{ span: 16 }}
            name="remember"
            valuePropName="checked"
            rules={[
              {
                validator: (_, value) =>
                  // eslint-disable-next-line prefer-promise-reject-errors
                  value
                    ? Promise.resolve()
                    : // eslint-disable-next-line prefer-promise-reject-errors
                      Promise.reject("请阅读并同意协议"),
              },
            ]}
          >
            <Checkbox>
              <span style={{ marginRight: 2 }}>我已阅读并同意</span>
              <a
                href="https://account.glodon.com/agreement"
                target="_blank"
                rel="noreferrer"
              >
                广联达用户协议
              </a>
              <span style={{ margin: "0 2px" }}>和</span>
              <a
                href="https://account.glodon.com/privacy"
                target="_blank"
                rel="noreferrer"
              >
                隐私政策
              </a>
            </Checkbox>
          </Form.Item>

          {onNext && (
            <Form.Item wrapperCol={{ span: 24 }}>
              <Button type="primary" htmlType="submit">
                下一步
              </Button>
            </Form.Item>
          )}
        </div>
      </Form>
    </>
  );
};
export default forwardRef<FormInstance, VerifyTelProps>(VerifyTel);
