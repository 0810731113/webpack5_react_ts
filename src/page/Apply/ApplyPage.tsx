import React, {
  useEffect,
  useState,
  useContext,
  createContext,
  useRef,
} from "react";
import { Switch, Route, useRouteMatch } from "react-router";
import { EnterpriseAccount, MobileBindAccount } from "api-auth/generated/model";
import {
  CheckCircleOutlined,
  ExclamationCircleFilled,
} from "@ant-design/icons";

import { Link } from "react-router-dom";
import { Steps, Button, message, Result, FormInstance, Space } from "antd";
import { useImmer } from "use-immer";
import "./ApplyPage.scss";
import consts from "consts";
import SystemHeader from "page/ProjectPage/ProjectPageComponents/SystemHeader";
import { IsPC } from "function/common.func";
import { ApplyParams } from "model/route-params.model";
import { onResponseError } from "function/auth.func";
import { applyService } from "service";
import { publishEvent } from "function/stats.func";
import VerifyTel from "./VerifyTel";
import VerifyAccount from "./VerifyAccount";
import Register from "./Register";

const { PUBLIC_URL, NESTBIM_BASE_URL } = consts;

interface ApplyPageProps {}

interface State {
  // accounts: MobileBindAccount[];
  // isNew: boolean;
  // applyingAccount: AccountInfo;
  // succeeded: boolean;
}

const { Step } = Steps;
export const ApplyPageContext = createContext<ApplyContext>({} as any);

export default function ApplyPage(props: ApplyPageProps) {
  const {} = props;
  const { succeeded, setSucceed } = useContext(ApplyPageContext);
  const registerFormRef = useRef<FormInstance>(null);
  const verifyTelFormRef = useRef<FormInstance>(null);
  const {
    params: { applyType },
  } = useRouteMatch<ApplyParams>();
  useEffect(() => {
    const changeFavicon = (link: string) => {
      let $favicon: any = document.querySelector('link[rel="shortcut icon"]');
      // If a <link rel="icon"> element already exists,
      // change its href to the given link.
      if ($favicon !== null) {
        $favicon.href = link;
        // Otherwise, create a new element and append it to <head>.
      } else {
        $favicon = document.createElement("link");
        $favicon.rel = "shortcut icon";
        $favicon.href = link;
        document.head.appendChild($favicon);
      }
    };
    changeFavicon(`${PUBLIC_URL}/design-favicon.ico`);
  }, []);
  useEffect(() => {
    publishEvent(`apply`, ["基础"], {
      eventLevel: "P1",
      accountType: applyType === "enterprise" ? "企业账号" : "个人账号",
    });
  }, []);

  const [current, setCurrent] = useState(0);

  const steps = [
    {
      title: "输入手机号码",
      description: "请输入您要绑定试用资格的手机号码",
      content: <VerifyTel onNext={() => setCurrent(1)} />,
    },
    {
      title: "账号名/密码设置",
      description: "请设置您的账号名和密码",
      content: <VerifyAccount onNext={() => setCurrent(2)} />,
    },
    {
      title: "填写申请信息",
      description: "继续广联达设计之旅",
      content: <Register />,
    },
  ];

  const individualCommit = async () => {
    const registerValues = await registerFormRef?.current?.validateFields();
    const verifyTelValues = await verifyTelFormRef?.current?.validateFields();
    try {
      const result = await applyService.reviewPersonalAccount({
        certificateUrl: registerValues.certificate,
        company: registerValues.company,
        department: registerValues.department,
        email: registerValues.passwordEmail,
        name: registerValues.userName,
        position: registerValues.position,
        qq: registerValues.qq,
        reason: registerValues.reason,
        telephone: verifyTelValues.mobile,
        verifyCode: verifyTelValues.code,
      });
      setSucceed();
    } catch (error) {
      if ([10002, 50007].includes(error.response?.data?.code)) {
        verifyTelFormRef?.current?.setFields([
          { errors: [error.response?.data?.msg], name: "code" },
        ]);
      } else {
        onResponseError(error);
      }
    }
  };
  const bannerPNG = `${
    applyType === "enterprise" ? "main_banner" : "personal_banner"
  }.png`;
  const headerStyle = {
    backgroundImage: `url("${PUBLIC_URL}/${bannerPNG}")`,
    backgroundColor: applyType === "enterprise" ? "#fff" : "#36565e",
  };
  return (
    <div className="apply-page">
      {succeeded ? (
        <>
          <div className="header" style={headerStyle}>
            <div className="slogon">
              <div className="main">一个账号，使用广联达所有产品</div>
              <div className="sub">
                AECORE 、构件坞 、BIMFACE 、广材网 、服务新干线 、协筑
                、CAD快速看图 、造价云等...
              </div>
            </div>
          </div>

          <div className="content succeeded">
            <Result
              title="提交成功"
              icon={<CheckCircleOutlined />}
              status="success"
              extra={
                <Space direction="vertical">
                  我们将尽快和您电话沟通，邮件通知您申请结果，请耐心等待
                  <a href={NESTBIM_BASE_URL}>
                    <Button
                      shape="round"
                      type="primary"
                      style={{ width: 200, fontSize: 16, marginTop: 16 }}
                      size="large"
                    >
                      返回官网
                    </Button>
                  </a>
                </Space>
              }
            />
          </div>
        </>
      ) : (
        <>
          {applyType === "enterprise" && (
            <div
              className="header"
              style={{
                backgroundImage: `url("${PUBLIC_URL}/main_banner.png")`,
              }}
            >
              <div className="steps">
                <Steps current={current}>
                  {steps.map((item) => (
                    <Step
                      key={item.title}
                      title={item.title}
                      description={
                        <span style={{ fontSize: 12 }}>{item.description}</span>
                      }
                    />
                  ))}
                </Steps>
              </div>
            </div>
          )}
          <div className="content">
            {([0, 2].includes(current) || applyType === "individual") && (
              <div className="exclamation">
                <ExclamationCircleFilled
                  style={{ color: "#faad14", marginRight: 5 }}
                />
                {applyType === "individual" || current === 0
                  ? "本产品目前处于邀请体验阶段，仅授权部分以住宅施工图设计为主要业务的建筑设计企业用户参与试用体验。感谢理解与支持！"
                  : "真实完整的申请信息有助于提高申请通过概率："}
              </div>
            )}
            {applyType === "enterprise" ? (
              steps[current].content
            ) : (
              <>
                <Register individual ref={registerFormRef} />
                <VerifyTel ref={verifyTelFormRef} />
                <Button
                  type="primary"
                  style={{ margin: "16px 0 32px" }}
                  onClick={individualCommit}
                >
                  提交申请
                </Button>
              </>
            )}
          </div>
        </>
      )}
    </div>
  );
}

export interface AccountInfo {
  identity?: string;
  password?: string;
  passwordMobile?: string;
  passwordEmail?: string;
  userName?: string;
  qq?: string;
  company?: string;
}

interface ApplyContext {
  accounts: MobileBindAccount[];
  isNew: boolean;
  succeeded: boolean;
  applyingAccount: AccountInfo | null;
  setAccounts: (accounts: MobileBindAccount[]) => void;
  setIsNew: (isNew: boolean) => void;
  setSucceed: () => void;
  setApplyingAccountInfo: (
    key:
      | "identity"
      | "password"
      | "passwordMobile"
      | "passwordEmail"
      | "userName"
      | "qq"
      | "company",
    value: string,
  ) => void;
}
