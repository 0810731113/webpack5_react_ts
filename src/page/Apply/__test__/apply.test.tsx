import React from "react";
import pretty from "pretty";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { render as renderHtml, unmountComponentAtNode } from "react-dom";
import { MobileBindAccount } from "api-auth/generated/model";

import ApplyPage, { ApplyPageContext } from "../ApplyPage";
import VerifyTel from "../VerifyTel";
import VerifyAccount from "../VerifyAccount";

let container: HTMLDivElement;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  if (container) {
    unmountComponentAtNode(container);
    container.remove();
  }
});

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
    replace: jest.fn(),
  }),
}));

test("test verify tel", () => {
  act(() => {
    renderHtml(<VerifyTel onNext={() => null} />, container);
  });

  expect(pretty(container?.innerHTML)).toMatchSnapshot();
});

test("test verify account", () => {
  const accounts = [
    {
      email: "",
      enterpriseUser: true,
      fullname: "SuperMAO",
      id: "6741899231286796949",
      passwordMobile: "15121143191",
      status: -1,
      username: "SuperMAO",
    },
  ];

  // render(
  //   <ApplyPageContext.Provider value={{ accounts }}>
  //     <VerifyAccount onNext={() => null} />
  //   </ApplyPageContext.Provider>,
  //   container,
  // );

  const defaultContext = {
    accounts,
    isNew: false,
    succeeded: false,
    applyingAccount: null,

    setSucceed: () => null,
    setAccounts: (newValue: MobileBindAccount[]) => null,
    setIsNew: jest.fn(),
    setApplyingAccountInfo: jest.fn(),
  };

  act(() => {
    renderHtml(
      <ApplyPageContext.Provider value={defaultContext}>
        <VerifyAccount onNext={() => null} />
      </ApplyPageContext.Provider>,
      container,
    );
  });

  expect(screen.getByTitle("SuperMAO")).toBeInTheDocument();
});
