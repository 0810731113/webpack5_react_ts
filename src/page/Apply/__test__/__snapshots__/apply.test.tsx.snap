// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test verify tel 1`] = `
"<div class=\\"content\\">
  <form id=\\"basic\\" class=\\"ant-form ant-form-horizontal\\">
    <div class=\\"exclamation\\"><span role=\\"img\\" aria-label=\\"exclamation-circle\\" style=\\"color: rgb(250, 173, 20); margin-right: 5px;\\" class=\\"anticon anticon-exclamation-circle\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"exclamation-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z\\"></path></svg></span>本产品目前处于邀请体验阶段，仅授权部分以住宅施工图设计为主要业务的建筑设计企业用户参与试用体验。感谢理解与支持！</div>
    <div class=\\"ant-row ant-form-item\\">
      <div class=\\"ant-col ant-col-5 ant-form-item-label ant-form-item-label-left\\"><label for=\\"basic_mobile\\" class=\\"ant-form-item-required\\" title=\\"手机号码\\">手机号码</label></div>
      <div class=\\"ant-col ant-col-19 ant-form-item-control\\">
        <div class=\\"ant-form-item-control-input\\">
          <div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"请输入有效的手机号码\\" type=\\"text\\" id=\\"basic_mobile\\" class=\\"ant-input ant-input-lg\\" value=\\"\\"></div>
        </div>
      </div>
    </div>
    <div class=\\"ant-row ant-form-item\\">
      <div class=\\"ant-col ant-col-5 ant-form-item-label ant-form-item-label-left\\"><label for=\\"basic_code\\" class=\\"ant-form-item-required\\" title=\\"验证码\\">验证码</label></div>
      <div class=\\"ant-col ant-col-19 ant-form-item-control\\">
        <div class=\\"ant-form-item-control-input\\">
          <div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg\\"><input placeholder=\\"请输入验证码\\" type=\\"text\\" id=\\"basic_code\\" class=\\"ant-input ant-input-lg\\" value=\\"\\"><span class=\\"ant-input-suffix\\"><a>获取验证码</a></span></span></div>
        </div>
      </div>
    </div>
    <div class=\\"commit-wrap\\">
      <div class=\\"ant-row ant-form-item\\">
        <div class=\\"ant-col ant-col-19 ant-form-item-control\\">
          <div class=\\"ant-form-item-control-input\\">
            <div class=\\"ant-form-item-control-input-content\\"><label class=\\"ant-checkbox-wrapper\\"><span class=\\"ant-checkbox\\"><input id=\\"basic_remember\\" type=\\"checkbox\\" class=\\"ant-checkbox-input\\" value=\\"\\"><span class=\\"ant-checkbox-inner\\"></span></span><span><span style=\\"margin-right: 2px;\\">我已阅读并同意</span><a href=\\"https://account.glodon.com/agreement\\" target=\\"_blank\\" rel=\\"noreferrer\\">广联达用户协议</a><span style=\\"margin: 0px 2px;\\">和</span><a href=\\"https://account.glodon.com/privacy\\" target=\\"_blank\\" rel=\\"noreferrer\\">隐私政策</a></span></label></div>
          </div>
        </div>
      </div>
      <div class=\\"ant-row ant-form-item\\">
        <div class=\\"ant-col ant-col-24 ant-form-item-control\\">
          <div class=\\"ant-form-item-control-input\\">
            <div class=\\"ant-form-item-control-input-content\\"><button type=\\"submit\\" class=\\"ant-btn ant-btn-primary\\"><span>下一步</span></button></div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>"
`;
